lib/patch/listener.ex

```diff
@@ -0,0 +1,111 @@
+defmodule Patch.Listener do
+  use GenServer
```

Comment:
would we want to use a child restart strategy of `:temporary` for tests? I might create a listener on a named process but accidentally crash it, it might get restarted by its own supervisor depending on its restart strategy (like if it's our etcd watcher or something which is supposed to be always running) and the listener could also get restarted by its own supervisor and re-initialize on the newly restarted target pid but then the listener doesn't have any of the call info. would it be clearer if any crash just caused the listener to die? curious what your thoughts are, if you already thought this through. I don't think I explained very well lol hope my question makes sense

Reply:
Hmm.... I had them supervised just so they didn't leak between tests but I hadn't considered this.  Let me noodle it some.

Reply:
Listeners are now `restart: :temporary` and there are a bunch more tests making sure they behave themselves.